### PR TITLE
Autofill visible password fields only

### PR DESCRIPTION
**Pull request by Gerard-CK**
Some site have their Register form, Reset Password form etc... as hidden panels. So only select the first visible password field when autofilling from Vault

### CODE REVIEW

**Comment by afteroot** [comment]
Interesting, will take a look at this

File: content.js [hunk 1]
```diff
@@ -92,9 +92,18 @@ function fillIn(node, value) {
   node.blur();
 }
 
+function findPasswordInput() {
+  const passwordNodes = document.querySelectorAll("input[type='password']");
+  for (let node of passwordNodes) {
+    if (node.offsetParent) return node;
```
Comment (mulbc) [inline content.js:98]: I'm not sure I understand the offsetParent function correctly, but how would that help in identifying visible inputs?

File: content.js [hunk 2]
```diff
@@ -92,9 +92,18 @@ function fillIn(node, value) {
...
+  }
+
+  return null;
+}
+
 function handleFillCredits(request) {
   // eslint-disable-next-line quotes
```
Comment (mulbc) [inline content.js:105]: if you move the querySelector, you have to move the eslint exception too

Comment (Gerard-CK) [reply content.js]: Indeed I do.

**Review by mulbc — CHANGES_REQUESTED** [review]
This looks very good already... but how about we just select the first node we found (if we found any) in case none passes our new node.offsetParent test?
That way we have a fallback to the previous way of working

Also why did you make the changes on lines 46-49?

**Comment by Gerard-CK** [comment]
> This looks very good already... but how about we just select the first node we found (if we found any) in case none passes our new node.offsetParent test? That way we have a fallback to the previous way of working

>> Yes, that makes sense. I'll work on that

> Also why did you make the changes on lines 46-49?
>> I realised during testing that some sites wrap their username field into a div or other container and give that an id of 'user', 'userid' etc... so I adjusted the query to pick input fields only otherwise we won't be able to set any value

**Review by mulbc — APPROVED** [review]
lgtm - thanks for working on this!

**Pull request merged**